src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -147,7 +147,28 @@ export default function Inbox ({ history: { push } }) {
   const refundAllDeclinedTransactions = useRefundAllDeclinedTransactions()
   const [counterpartyNotFound, setCounterpartyNotFound] = useState(true)
   const [isDeclinedTransactionModalVisible, setIsDeclinedTransactionModalVisible] = useState(false)
-  const [modalTransaction, setModalTransaction] = useState(null)
+  const [confirmationModalProperties, setConfirmationModalProperties] = useState({
+    shouldDisplay: false,
+    transactions: [],
+    action: '',
+    hasConfirmed: false
+  })
+
+  const resetDefaultModalTransaction = () => {
+    setConfirmationModalProperties({
+      shouldDisplay: false,
+      transactions: [],
+      action: '',
+      hasConfirmed: false
+    })
+  }
+
+  const setNewModalTransactionProperties = (newProperties = {}) => {
+    if (isEmpty(newProperties)) resetDefaultModalTransaction()
+    else setConfirmationModalProperties(newProperties)
+  }
```

Comment:
I'm hoping we can avoid this function and just use `setConfirmationModalProperties`.

Reply:
The way that our whole component is written has extracted the zome call functionality into the modals; therefore the knowledge of which transaction has been action-ed lives in the modals, which is a sibling of `TransactionRow`, where each of the transactions live.  

I'm happy to pair on this though...